cmd/podman/ps.go

```diff
@@ -205,9 +205,19 @@ func checkFlagsPassed(c *cliconfig.PsValues) error {
 	if c.Last >= 0 && c.Latest {
 		return errors.Errorf("last and latest are mutually exclusive")
 	}
-	// Filter forces all
+	// Filter on status forces all
 	if len(c.Filter) > 0 {
-		c.All = true
+		foundStatus := false
+		for _, filter := range c.Filter {
+			splitFilter := strings.SplitN(filter, "=", 2)
+			if strings.ToLower(splitFilter[0]) == "status" {
+				foundStatus = true
+				break
+			}
+		}
+		if foundStatus {
+			c.All = true
```

Comment:
No need for foundStatus, just set the c.All=true above in the foundStatus=true line.

Reply:
Fixed